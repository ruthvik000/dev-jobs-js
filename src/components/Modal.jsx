import React from 'react'

export default function Modal({open, children, onClose}) {
    if (!open) return null
    return (
        <>
            <div className='overlay-popup'/>
            <div className='modal-styles'>
                {children}
                <div>
                    <button onClick={onClose} type='button'>Close</button>
                </div>
            </div>
        </>
    )
}