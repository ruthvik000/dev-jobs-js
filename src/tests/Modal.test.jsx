import { shallow } from 'enzyme';
import Modal from "../components/Modal";

describe('Modal component', () => {
    it('should renders div with close modal', () => {
        const container = shallow(<Modal open={true} />);
        expect(container.html()).toMatchSnapshot();
    });

    it('should render null', () => {
        const container = shallow(<Modal />);
        expect(container.html()).toMatchSnapshot();
    });
});