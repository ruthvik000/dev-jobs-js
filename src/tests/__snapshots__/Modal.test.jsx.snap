// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render null 1`] = `null`;

exports[`Modal component should renders div with close modal 1`] = `"<div class=\\"overlay-popup\\"></div><div class=\\"modal-styles\\"><div><button type=\\"button\\">Close</button></div></div>"`;
